src/components/SearchBar/SearchBar.component.jsx

```diff
@@ -1,14 +1,15 @@
 import React from 'react';
 import { faSearch } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useSearch } from '../../providers/Search';
 import { SearchBarWrapper, SearchBarIcon, SearchBarInput } from './SearchBar.styled';
 
-function SearchBar(props) {
-  const { search } = props;
+function SearchBar() {
+  const { setSearch } = useSearch();
 
   const inputSearch = (e) => {
     if (e.key === 'Enter') {
-      search(e.target.value);
+      setSearch(e.target.value);
```

Comment:
you might want to set it also in the input so even if you change pages, the user always knows what they are searching for.